packages/salesforce/test/Adaptor.test.js

```diff
@@ -194,4 +195,86 @@ describe('Adaptor', () => {
       expect(result).to.eql('Nhamaonha 6a Classe 2023-10-09');
     });
   });
+
+  describe('query', () => {
+    it('should fetch all records', done => {
+      const fakeConnection = {
+        query: function () {
+          return Promise.resolve({
+            done: true,
+            totalSize: 1,
+            records: [{ Name: 'OpenFn' }],
+          });
+        },
+      };
+      let state = { connection: fakeConnection, references: [] };
+      let spy = sinon.spy(fakeConnection, 'query');
+
+      query('select Name from Account', { autoFetch: true })(state)
+        .then(state => {
+          expect(spy.called).to.eql(true);
+          expect(state.references[0]).to.eql({
+            done: true,
+            totalSize: 1,
+            records: [{ Name: 'OpenFn' }],
+          });
+        })
+        .then(done)
+        .catch(done);
+    });
+    it('should fetch all records if autoFetch is true and totalSize > 2000', done => {
+      const fakeConnection = {
+        query: function () {
+          return Promise.resolve({
+            done: true,
+            totalSize: 5713,
+            records: [{ Name: 'OpenFn' }],
+          });
+        },
+      };
+      let state = { connection: fakeConnection, references: [] };
+      let spy = sinon.spy(fakeConnection, 'query');
+
+      query('select Name from Account', { autoFetch: true })(state)
+        .then(state => {
+          expect(spy.called).to.eql(true);
+          expect(state.references[0]).to.eql({
+            done: true,
+            totalSize: 5713,
+            records: [{ Name: 'OpenFn' }],
+          });
+        })
+        .then(done)
+        .catch(done);
+    });
+    it('should return nextRecordsUrl if autoFetch is false and totalSize > 2000', done => {
```

Comment:
I think the test name should be more like `should not fetch another page if autofetch is false`. Because that explains the behaviour, not the implementation details